lib/pacemaker/pcmk_ticket.c

```diff
@@ -146,3 +147,67 @@ pcmk_ticket_get_attr(xmlNodePtr *xml, const char *ticket_id,
     pe_free_working_set(scheduler);
     return rc;
 }
+
+int
+pcmk__ticket_info(pcmk__output_t *out, pcmk_scheduler_t *scheduler,
+                  const char *ticket_id, bool details, bool raw)
+{
+    int rc = pcmk_rc_ok;
+
+    CRM_ASSERT(out != NULL && scheduler != NULL);
+
+    if (ticket_id != NULL) {
+        GHashTable *tickets = NULL;
+        pcmk_ticket_t *ticket = g_hash_table_lookup(scheduler->tickets, ticket_id);
+
+        if (ticket == NULL) {
+            return ENXIO;
+        }
+
+        /* The ticket-list message expects a GHashTable, so we'll construct
+         * one with just this single item.
+         */
+        tickets = pcmk__strkey_table(free, NULL);
+        g_hash_table_insert(tickets, strdup(ticket->id), ticket);
+        out->message(out, "ticket-list", tickets, false, raw, details);
+        g_hash_table_destroy(tickets);
+
+    } else {
+        out->message(out, "ticket-list", scheduler->tickets, false, raw, details);
+    }
+
+    return rc;
+}
+
+int
+pcmk_ticket_info(xmlNodePtr *xml, const char *ticket_id)
+{
+    pcmk_scheduler_t *scheduler = NULL;
+    pcmk__output_t *out = NULL;
+    int rc = pcmk_rc_ok;
+
```

Comment:
check for invalid args

Reply:
Handled by `pcmk__xml_output_new` here too.